File: Exercism/csharp/grains/Grains.cs
using System;
using System.Linq;

public static class Grains
{
    public static ulong Square(int n) => n > 0 && n < 65
      ? (ulong)(1UL << n - 1)
      : throw new ArgumentOutOfRangeException();

    public static ulong Total() => (ulong)Enumerable.Range(1, 64).Select(i => (long)Square(i)).Sum();
}

File: Exercism/fsharp/grains/Grains.fs
module Grains

let square = function
    | n when n < 1 || n > 64 -> "square must be between 1 and 64" |> Error
    | n -> (n - 1) |> pown 2UL |> Ok

let total: Result<uint64, string> =
    [ 1 .. 64 ]
    |> Seq.fold (fun acc it -> acc + pown 2UL (it - 1)) 0UL
    |> Ok